pkg/provider/azure_vmss.go

```diff
@@ -1467,25 +1464,94 @@ func (ss *ScaleSet) ensureHostsInPool(ctx context.Context, service *v1.Service,
 				klog.ErrorS(err, "Failed to get vmss batch size", logFields...)
 				return err
 			}
-
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+			errChan := make(chan error, len(update))
+			pollerChannel := make(chan *runtime.Poller[armcompute.VirtualMachineScaleSetVMsClientUpdateResponse], len(update))
+			var pollerGroup sync.WaitGroup
+			pollerGroup.Add(1)
+			go func() {
+				defer pollerGroup.Done()
+				for {
+					select {
+					case poller, ok := <-pollerChannel:
+						if !ok {
+							// pollerChannel is closed
+							return
+						}
+						if poller == nil {
```

Comment:
would this happen? when codes go here, that means there are something in pollerChannel.